src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/leaked-passwords/LeakedPasswordsLayout.tsx

```diff
@@ -18,76 +19,193 @@ import {
   StepDeterminationData,
   StepLink,
   getNextGuidedStep,
+  hasCompletedStep,
 } from "../../../../../../../functions/server/getRelevantGuidedSteps";
 import { FixView } from "../FixView";
 import { getGuidedExperienceBreaches } from "../../../../../../../functions/universal/guidedExperienceBreaches";
+import {
+  BreachResolutionRequest,
+  HibpBreachDataTypes,
+} from "../../../../../../../(nextjs_migration)/(authenticated)/user/breaches/breaches";
+import { useEffect, useState } from "react";
 
 export interface LeakedPasswordsLayoutProps {
   type: LeakedPasswordsTypes;
   subscriberEmails: string[];
   data: StepDeterminationData;
 }
 
-export function LeakedPasswordsLayout(props: LeakedPasswordsLayoutProps) {
-  const l10n = useL10n();
+async function updateBreachStatus(
+  email: string,
+  id: number,
+  resolvedDataClass: HibpBreachDataTypes[keyof HibpBreachDataTypes],
+) {
+  try {
+    const data: BreachResolutionRequest = {
+      affectedEmail: email,
+      breachId: id,
+      resolutionsChecked: [resolvedDataClass],
```

Comment:
Does `resolvedDataClass` include all the existing checked resolutions? If not this will be an problem since it'll replace the current resolutions that are checked.

Reply:
Looks like it [only contains the one that has been just resolved](https://github.com/mozilla/blurts-server/pull/3712/files#diff-804c73dc7ceb2ac0c19bf0f0f4b0cf6d67decba2aa26c0764b70c9bf310a8eb3R95) — so this would need to be updated.